crates/core/tedge_agent/src/agent.rs

```diff
@@ -433,10 +434,10 @@ pub fn create_tedge_to_te_converter(
 
     // Tedge to Te converter
     let mut tedge_converter_actor =
-        ConvertingActor::builder("TedgetoTeConverter", tedge_to_te_converter, subscriptions);
+        ConvertingActor::builder("TedgetoTeConverter", tedge_to_te_converter);
 
-    tedge_converter_actor.add_input(mqtt_actor_builder);
-    tedge_converter_actor.add_sink(mqtt_actor_builder);
+    tedge_converter_actor.add_input(subscriptions, mqtt_actor_builder);
```

Comment:
```suggestion
    mqtt_actor_builder.add_sink(subscriptions, &tedge_converter_actor);
```
This is the counter-intuitive bit that I was talking about earlier. This variant is clearer to me than the reversed version. But, this is just my personal view.

Reply:
Indeed, a personal view ;-) I prefer the reverse, .i.e. adding a source.